libs/blocks/redirects-formatter/utils/redirect-inputs.js

```diff
@@ -0,0 +1,101 @@
+  <defs>
+    <style>
+      .fill {
+        fill: #464646;
+      }
+    </style>
+  </defs>
+  <title>S Delete 18 N</title>
+  <rect id="Canvas" fill="#ff13dc" opacity="0" width="18" height="18" /><path class="fill" d="M15.75,3H12V2a1,1,0,0,0-1-1H6A1,1,0,0,0,5,2V3H1.25A.25.25,0,0,0,1,3.25v.5A.25.25,0,0,0,1.25,4h1L3.4565,16.55a.5.5,0,0,0,.5.45H13.046a.5.5,0,0,0,.5-.45L14.75,4h1A.25.25,0,0,0,16,3.75v-.5A.25.25,0,0,0,15.75,3ZM5.5325,14.5a.5.5,0,0,1-.53245-.46529L5,14.034l-.5355-8a.50112.50112,0,0,1,1-.067l.5355,8a.5.5,0,0,1-.46486.53283ZM9,14a.5.5,0,0,1-1,0V6A.5.5,0,0,1,9,6ZM11,3H6V2h5Zm1,11.034a.50112.50112,0,0,1-1-.067l.5355-8a.50112.50112,0,1,1,1,.067Z" />
+</svg>`;
+const addIcon = `<svg xmlns="http://www.w3.org/2000/svg" height="18" viewBox="0 0 18 18" width="18">
+<defs>
+  <style>
+    .fill {
+      fill: #464646;
+    }
+  </style>
+</defs>
+<title>S Add 18 N</title>
+<rect id="Canvas" fill="#ff13dc" opacity="0" width="18" height="18" /><path class="fill" d="M14.5,8H10V3.5A.5.5,0,0,0,9.5,3h-1a.5.5,0,0,0-.5.5V8H3.5a.5.5,0,0,0-.5.5v1a.5.5,0,0,0,.5.5H8v4.5a.5.5,0,0,0,.5.5h1a.5.5,0,0,0,.5-.5V10h4.5a.5.5,0,0,0,.5-.5v-1A.5.5,0,0,0,14.5,8Z" />
+</svg>`;
+
+function createSingleInputRow(id) {
+  const fromLabel = createTag('label', { for: `source-${id}` }, 'Source URL');
+  const fromInput = createTag('input', { class: 'source', type: 'text', id: `source-${id}`, name: `source-${id}` });
+  const toLabel = createTag('label', { for: `destination-${id}` }, 'Destination URL');
+  const toInput = createTag('input', { class: 'destination', type: 'text', id: `destination-${id}`, name: `destination-${id}` });
+  const removeInputButton = createTag('button', { class: 'remove-input', 'data-input-id': id }, deleteIcon);
+
+  const inputContainer = createTag('div', { class: 'input-container' }, [fromLabel, fromInput, toLabel, toInput]);
+
+  removeInputButton.addEventListener('click', () => {
+    const removalId = removeInputButton.dataset.inputId;
+    const inputsParent = document.querySelector('.single-redirects-container');
+    const elementToRemove = inputsParent.querySelector(`.redirect-input-row[data-row-id="${removalId}"]`);
+    elementToRemove.remove();
+  });
+
+  return createTag('div', { class: 'redirect-input-row', 'data-row-id': id }, [inputContainer, removeInputButton]);
+}
+
+export default function createSingleRedirectArea() {
+  const singleInput = createSingleInputRow(startingRowId);
+  const addInputButton = createTag('button', { class: 'add-input' }, addIcon);
+  const addSingleInput = createTag('div', { class: 'single-input-ui' }, addInputButton);
+
+  addSingleInput.addEventListener('click', () => {
+    let newRowId = startingRowId;
+    const currentRows = document.querySelectorAll('.redirect-input-row');
+    const numRows = currentRows.length;
+
+    if (numRows >= 1) {
+      const lastRowId = currentRows[numRows - 1].dataset.rowId;
+      newRowId = parseInt(lastRowId, 10) + 1;
+    }
```

Comment:
Did you consider using a container data attribute or a scoped variable to track the next rowId?

Reply:
That is a good question. I think I went with this pattern to use the DOM as state. I think that if I added the row id as a data attribute, I'd just be changing the data attribute and querying it instead of querying the last item in the current DOM. Do you think it would be more performant?

Reply:
It should be a little more performant, you wouldn't have to query and check the number of rows, just set `nextRowId` to 1 when you set up the container, or use `lastRowId = 0` if you prefer.
```js
const redirectContainer = document.querySelector('.single-redirects-container')
newRowId = parseInt(redirectContainer.dataset.nextRowId, 10)
const newRow = createSingleInputRow(newRowId);
redirectContainer.insertBefore(newRow, addSingleInput);
redirectContainer.dataset.nextRowId = newRowId + 1;;
```